### PR TITLE
[ETP-292] ar form overrides page content

### PR DESCRIPTION
**Pull request by Luc45**
### Ticket
https://moderntribe.atlassian.net/browse/ETP-292

### What was happening:
- The page content was being spoofed (set to an empty string) [here](https://github.com/moderntribe/event-tickets/blob/release/G20.04/src/Tribe/Attendee_Registration/Template.php#L64)
- Then we were filtering the content to show the contents of the Attendee Registration page, [here](https://github.com/moderntribe/event-tickets/blob/release/G20.04/src/Tribe/Attendee_Registration/Template.php#L269)

Which resulted in the `global $post->post_content` being replaced with the return of `\Tribe__Tickets__Attendee_Registration__View::display_attendee_registration_page`.

### How it was solved
- We don't spoof the page anymore if the shortcode is in the content
- We don't filter `the_content` anymore if the shortcode is in the content

### Other modifications
While writing and running the functional tests, I was getting some fatal errors and warnings from the templates, trying to "call a method on a boolean" or "trying to read from a non-object". Turns out they rely on the `$provider` variable to be an actual provider, but the function that provides it can also return false.

I don't know if there's any scenario where this could happen to a real client or be replicable without tweaking the inner workings or doing something weird. But, just in case, I've added some sanity checks before trying to read from that variable. If it's not an object, just default to an empty string.

### CODE REVIEW

File: src/Tribe/Attendee_Registration/Template.php [hunk 1]
```diff
@@ -48,6 +48,8 @@ public function hook() {
 	 *
 	 * @since 4.9
 	 *
+	 * @param WP_Post  $posts Post data object.
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php:51]: :no_entry_sign: Expected 1 spaces after parameter type; 2 found

<sub>hash: 9d31eee5e92badb3c636512ab0dd806c</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 2]
```diff
@@ -58,15 +60,28 @@ public function setup_context( $posts ) {
...
 		}
 
+		/*
+		 * Early bail:
+		 * We are on the AR page, but we have the shortcode in the content,
+		 * so we don't want to spoof this page.
+		 */
+		if ( is_array( $posts ) && ! empty( $posts ) ) {
+			if ( $posts[0] instanceof WP_Post ) {
+				if ( has_shortcode( $posts[0]->post_content, "tribe_attendee_registration" ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php:70]: :no_entry_sign: String "tribe_attendee_registration" does not require double quotes; use single quotes instead

<sub>hash: 961656190d56c435d2ccc17a2ae8b838</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 3]
```diff
@@ -262,11 +277,17 @@ public function set_page_content( $query ) {
 		}
 
 		if ( $this->is_main_loop( $query ) ) {
+			global $post;
+			if ( $post instanceof WP_Post && has_shortcode( $post->post_content, "tribe_attendee_registration" ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php:281]: :no_entry_sign: String "tribe_attendee_registration" does not require double quotes; use single quotes instead

<sub>hash: 1a1f2b3be2b8839878d70a1419d6c8f7</sub>

File: src/views/blocks/tickets/footer-total.php [hunk 4]
```diff
@@ -27,6 +33,6 @@
 		<?php echo esc_html_x( 'Total:', 'Total selected tickets price.', 'event-tickets' ); ?>
 	</span>
 	<span class="tribe-tickets__footer__total__wrap">
-		<?php echo $tribe_commerce_currency->get_formatted_currency_with_symbol( 0, $post_id, $provider->class_name ); ?>
+		<?php echo $tribe_commerce_currency->get_formatted_currency_with_symbol( 0, $post_id, $provider_class); ?>
```
Comment (tr1b0t) [inline src/views/blocks/tickets/footer-total.php:36]: * :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tribe_commerce_currency'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$tribe_commerce_currency'.
* :no_entry_sign: Expected 1 spaces before closing bracket; 0 found
* :no_entry_sign: There must be a single space before the last parenthesis
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$post_id'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$provider_class'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$provider_class'.

<sub>hash: 6fb538f97b7f88039ac2c9609b142434</sub>

File: src/views/blocks/tickets/footer.php [hunk 5]
```diff
@@ -20,8 +20,16 @@
 $tickets         = $this->get( 'tickets' );
 $currency_symbol = $this->get( 'currency_symbol' );
 $provider        = $this->get( 'provider' );
-$cart_url        = $provider->get_cart_url();
-$checkout_url    = $provider->get_checkout_url();
+if ( method_exists( $provider, "get_cart_url" ) ) {
```
Comment (tr1b0t) [inline src/views/blocks/tickets/footer.php:23]: :no_entry_sign: String "get_cart_url" does not require double quotes; use single quotes instead

<sub>hash: 20462959028050a27ea029685345d1a9</sub>

File: src/views/blocks/tickets/footer.php [hunk 6]
```diff
@@ -20,8 +20,16 @@
...
+	$cart_url = $provider->get_checkout_url();
+} else {
+	$cart_url = '';
+}
+if ( method_exists( $provider, "get_cart_provider" ) ) {
```
Comment (tr1b0t) [inline src/views/blocks/tickets/footer.php:28]: :no_entry_sign: String "get_cart_provider" does not require double quotes; use single quotes instead

<sub>hash: 6bc7192ee9ce5897a4cb585d932591b7</sub>

File: src/views/registration-js/content.php [hunk 7]
```diff
@@ -28,6 +28,12 @@
 	$provider     = $provider_obj->attendee_object;
 }
 
+if ( method_exists( $provider_obj, "get_cart_provider" ) ) {
```
Comment (tr1b0t) [inline src/views/registration-js/content.php:31]: :no_entry_sign: String "get_cart_provider" does not require double quotes; use single quotes instead

<sub>hash: d82f83fdf9f29921dc688cc83404cd87</sub>

File: src/views/registration/button-cart.php [hunk 8]
```diff
@@ -16,7 +16,12 @@
 $provider     = $this->get( 'provider' );
 $cart_url     = $this->get_cart_url( $provider );
 $provider_obj = $this->get_cart_provider( $provider );
-$checkout_url = $provider_obj->get_checkout_url();
+
+if ( method_exists( $provider_obj, "get_cart_provider" ) ) {
```
Comment (tr1b0t) [inline src/views/registration/button-cart.php:20]: :no_entry_sign: String "get_cart_provider" does not require double quotes; use single quotes instead

<sub>hash: 5bf8d95ad6b4bcd28b159fa025106c02</sub>

File: src/Tribe/Attendee_Registration/Template.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/Tribe/Attendee_Registration/Template.php:70]:
```suggestion
				if ( has_shortcode( $posts[0]->post_content, 'tribe_attendee_registration' ) ) {
```

File: src/Tribe/Attendee_Registration/Template.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Attendee_Registration/Template.php:51]:
```suggestion
	 * @param WP_Post $posts Post data object.
```

File: src/Tribe/Attendee_Registration/Template.php (hunk 3 above)
Comment (sc0ttkclark) [reply src/Tribe/Attendee_Registration/Template.php:281]:
```suggestion
			if ( $post instanceof WP_Post && has_shortcode( $post->post_content, 'tribe_attendee_registration' ) ) {
```

File: src/views/blocks/tickets/footer.php (hunk 5 above)
Comment (sc0ttkclark) [reply src/views/blocks/tickets/footer.php:23]:
```suggestion
if ( method_exists( $provider, 'get_cart_url' ) ) {
```

File: src/views/blocks/tickets/footer.php (hunk 6 above)
Comment (sc0ttkclark) [reply src/views/blocks/tickets/footer.php:28]:
```suggestion
if ( method_exists( $provider, 'get_cart_provider' ) ) {
```

File: src/views/registration-js/content.php (hunk 7 above)
Comment (sc0ttkclark) [reply src/views/registration-js/content.php:31]:
```suggestion
if ( method_exists( $provider_obj, 'get_cart_provider' ) ) {
```

File: src/views/registration/button-cart.php (hunk 8 above)
Comment (sc0ttkclark) [reply src/views/registration/button-cart.php:20]:
```suggestion
if ( method_exists( $provider_obj, 'get_cart_provider' ) ) {
```

File: src/views/blocks/tickets/footer-total.php (hunk 4 above)
Comment (sc0ttkclark) [inline src/views/blocks/tickets/footer-total.php:36]:
```suggestion
		<?php echo $tribe_commerce_currency->get_formatted_currency_with_symbol( 0, $post_id, $provider_class ); ?>
```

File: src/views/blocks/tickets/footer-total.php [hunk 9]
```diff
@@ -27,6 +33,6 @@
 		<?php echo esc_html_x( 'Total:', 'Total selected tickets price.', 'event-tickets' ); ?>
 	</span>
 	<span class="tribe-tickets__footer__total__wrap">
-		<?php echo $tribe_commerce_currency->get_formatted_currency_with_symbol( 0, $post_id, $provider->class_name ); ?>
+		<?php echo $tribe_commerce_currency->get_formatted_currency_with_symbol( 0, $post_id, $provider_class ); ?>
```
Comment (tr1b0t) [inline src/views/blocks/tickets/footer-total.php:36]: * :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tribe_commerce_currency'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$tribe_commerce_currency'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$post_id'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$provider_class'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$provider_class'.

<sub>hash: c075af508885c6183a92ef80a914f6e1</sub>

File: src/views/blocks/tickets/footer-total.php [hunk 10]
```diff
@@ -27,6 +34,6 @@
...
```
Comment (tr1b0t) [inline src/views/blocks/tickets/footer-total.php:37]: * :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tribe_commerce_currency'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$tribe_commerce_currency'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$post_id'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$provider_class'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$provider_class'.

<sub>hash: c82d24767a811c6777805a714b798be7</sub>

Comment (tr1b0t) [inline src/views/blocks/tickets/footer-total.php:37]: * :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tribe_commerce_currency'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$tribe_commerce_currency'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$post_id'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$provider_class'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$provider_class'.

<sub>hash: c82d24767a811c6777805a714b798be7</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 11]
```diff
@@ -262,11 +277,17 @@ public function set_page_content( $query ) {
...
+				// Early bail: There's no need to override the content if the post is using the shortcode.
+				return;
+			}
+
 			// Prevent the TEC v2 page override from preventing our content override.
 			add_filter( 'tribe_events_views_v2_should_hijack_page_template', '__return_false' );
 
 			// Load Attendee Registration view for the content.
-			add_filter( 'the_content', array( tribe( 'tickets.attendee_registration.view' ), 'display_attendee_registration_page' ) );
+			add_filter( 'the_content', [ tribe( 'tickets.attendee_registration.view' ), 'display_attendee_registration_page' ] );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Template.php:290]:
```suggestion
			add_filter( 'the_content', tribe_callback( 'tickets.attendee_registration.view', 'display_attendee_registration_page' ) );
```

File: src/Tribe/Attendee_Registration/Template.php [hunk 12]
```diff
@@ -58,15 +60,28 @@ public function setup_context( $posts ) {
...
 		// Don't tell wp_query we're anything in particular - then we don't run into issues with defaults.
-		$wp_query->is_page     = false;
-		$wp_query->is_singular = false;
-		$wp_query->is_home     = false;
+		$wp_query->is_page        = false;
+		$wp_query->is_singular    = false;
+		$wp_query->is_home        = false;
 		$wp_query->is_archive  = false;
 		$wp_query->is_category = false;
 		$wp_query->is_404      = false;
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Template.php:87]:
```suggestion
		$wp_query->is_page     = false;
		$wp_query->is_singular = false;
		$wp_query->is_home     = false;
		$wp_query->is_archive  = false;
		$wp_query->is_category = false;
		$wp_query->is_404      = false;
```

File: src/Tribe/Attendee_Registration/Template.php [hunk 13]
```diff
@@ -48,6 +48,8 @@ public function hook() {
 	 *
 	 * @since 4.9
 	 *
+	 * @param WP_Post $posts Post data object.
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Template.php:51]:
```suggestion
	 * @param WP_Post[] $posts Post data objects.
```

File: readme.txt [hunk 14]
```diff
@@ -119,6 +119,7 @@ Currently, the following add-ons are available for Event Tickets:
 
 = [4.12.0] TBD =
 
+* Fix - Using a custom Attendee Registration page with the [tribe_attendee_registration] shortcode will no longer replace the contents of the page. [ETP-292]
```
Comment (sc0ttkclark) [inline readme.txt:122]:
```suggestion
* Fix - When using Event Tickets Plus and using a custom Attendee Registration page with the [tribe_attendee_registration] shortcode, Event Tickets will no longer replace the contents of the page. [ETP-292]
```

**Pull request merged**